pkg/cmd/initialize/bpfobject.go

```diff
@@ -15,110 +15,32 @@ import (
 	"github.com/aquasecurity/tracee/pkg/logger"
 )
 
+// BpfObject sets up and configures a BPF object for tracing and monitoring
+// system events within the kernel. It takes pointers to tracee.Config,
+// helpers.KernelConfig, and helpers.OSInfo structures, as well as an
+// installation path and a version string. The function unpacks the CO-RE eBPF
+// object binary, checks if BTF is enabled, unpacks the BTF file from BTF Hub if
+// necessary, and assigns the kernel configuration and BPF object bytes.
 func BpfObject(config *tracee.Config, kConfig *helpers.KernelConfig, OSInfo *helpers.OSInfo, installPath string, version string) error {
-	var d = struct {
-		btfenv     bool
-		bpfenv     bool
-		btfvmlinux bool
-	}{
-		btfenv:     false,
-		bpfenv:     false,
-		btfvmlinux: helpers.OSBTFEnabled(),
-	}
-
-	bpfFilePath, err := checkEnvPath("TRACEE_BPF_FILE")
-	if bpfFilePath != "" {
-		d.bpfenv = true
-	} else if bpfFilePath == "" && err != nil {
-		return errfmt.WrapError(err)
-	}
-	btfFilePath, err := checkEnvPath("TRACEE_BTF_FILE")
```

Comment:
Why ? What environments ? Why wouldn't we have those envs in BTFHUB ?

Reply:
1) Is Android a target distro for Tracee ? I remember we had that discussion in a recent past and said it wasn't and decisions should not be made because of that.

2) Unsupported distros are "unsupported" and should have BTF embedded (or continue to be unsupported ?).

3) Custom built kernels can always provide BTF info through kconfigs.

I got your point, but I still think its just some residual concern of dropping non CORE.

Reply:
With that said, I'm ok in keeping it.

Reply:
I have included it in commit:

commit 2d0dfd95
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Thu Apr 27 12:59:36 2023

    ebpf: re-introduce simpler external BTF file logic